pkg/base1/cockpit.js

```diff
@@ -2163,33 +2161,34 @@ function full_scope(cockpit, $, po) {
                 if (quotient < factor) {
                     number = quotient;
                     suffix = suffixes[factor][i];
-                    divided = divisor > 1;
                     break;
                 }
                 divisor *= factor;
             }
         }
 
-        var ret;
-
-        if (!suffix) {
-            ret = [number.toString()];
-            if (!separate)
-                ret = ret.join(" ");
-            return ret;
-        }
-
         /* non-zero values should never appear zero */
-        if (number > 0 && number < 0.1)
-            number = 0.1;
+            if (number > 0 && number < 0.1)
+                number = 0.1;
+            else if (number < 0 && number > -0.1)
+                number = -0.1;
+
+        var ret;
 
         /* TODO: Make the decimal separator translatable */
-        if (number === 0 || !divided)
-            ret = [number.toString(), suffix];
-        else
+        if (suffix)
             ret = [number.toFixed(1), suffix];
+        else
+            ret = [number.toFixed(1)];
```

Comment:
This changes the API. This API is meant to return two values unless ```!separate```. What's the logic behind this change. This change is also not documented.